### DISCUSSION TITLE
Evaluate strings like jq instead of files

### DISCUSSION THREAD

**Discussion by marcellodesales** [mikefarah/yq#706] [Q&A] [answered]
# `jq` uses input stream to evaluate strings

* With any Json, I can quickly parse/evaluate the string value...
 
```console
echo '{"name": "Marcello"}' | \
        docker run --rm -i imega/jq '.name == "Marcello"'
true
```

# `yq` requires a file path to evaluate

* My current solution with `yq` requires a file to evaluate values:

```console
echo 'name: Marcello'  > a.yaml  | \
       docker run --rm -i -v $(pwd):$(pwd) mikefarah/yq eval '.name == "Marcello"' $(pwd)/a.yaml
true
```

# Question: How can I evaluate yaml strings instead of files?

* How can I evaluate strings similar to` jq`???
  * As I understand, `jq` uses an input stream instead of the file path

**Comment by mikefarah** [comment, marked as answer]
You can pass in `-` as a filename to read from STDIN.

**Reply by marcellodesales** [reply]
@mikefarah Thank you! I found the reference and I didn't reload the page :) I posted a comment with the answer too :D

**Comment by marcellodesales** [comment]
# Answer 

* Oops... Just found the answer at https://mikefarah.gitbook.io/yq/v/v4.x/commands/evaluate#usage
  * Use `-` as the file name for `STDIN`

```console
$ echo 'name: Marcello' | docker run --rm -i -v $(pwd):$(pwd) mikefarah/yq eval '.name == "Marcello"' -
true
```
